azure-pipelines.yml

```
@@ -293,12 +298,14 @@ stages:
     dependsOn: Aks
     jobs:
       - job: BootStrap
+        condition: and(succeeded(), eq(variables['isMain'], true), eq('${{ parameters.action }}', 'apply'))
```

this doesn't make any sense nothing else (except tests for some reason with the same bug) have an is main check.

It's perfectly legit to rebuild clusters on a PR as we roll a change out. as long as by the time it gets to prod it gets merged. Ideally we would be merging and building on master but that may not always be possible and also there may be experimental work in sandbox, that should still cause bootstrap to be run and the issuer url to be updated.

The condition should be roughly this and is based on https://github.com/hmcts/cnp-azuredevops-libraries/blob/master/steps/terraform.yaml#L214C1-L218C8

also pipeline tests should be updated too
```suggestion
        condition: |
	    or(
	      and(succeeded(), eq('${{ parameters.action }}', 'apply')),
	      and(succeeded(), eq(variables['isMain'], true), eq(variables['isAutoTriggered'], true))
	    )
```

---

Sure - I only added it at this layer in the PR as it's already present in the internal steps below - so it was happening anyway before these changes, agreed on the reasoning to update this and pipeline tests though so will add the change

https://github.com/hmcts/aks-cft-deploy/blob/main/pipeline-steps/bootstrap.yaml#L16
https://github.com/hmcts/aks-cft-deploy/blob/main/pipeline-steps/bootstrap.yaml#L27

---

aware, meant to mention that, but it was wrong before and it's much worse now.

---

That should [be sorted now](https://github.com/hmcts/aks-cft-deploy/pull/537/commits/c1d04584846f697f095b424d6329ff28aa1776fe) - I think this is all you meant